features/vaadin-dashlets/dashlet-alarms/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/AlarmDetailsDashlet.java

```diff
@@ -274,9 +291,25 @@ private boolean checkBoosted(List<OnmsAlarm> alarms) {
         alarmCb.fetch("firstEvent", Fetch.FetchType.EAGER);
         alarmCb.fetch("lastEvent", Fetch.FetchType.EAGER);
 
-        alarmCb.distinct();
+        /**
+         * due to restrictions in the criteria api it's quite hard
+         * to use distinct and orderBy together, so I apply a workaround
+         * to avoid alarmCb.distinct();
+         */
+
+        List<OnmsAlarm> onmsAlarmList = m_alarmDao.findMatching(alarmCb.toCriteria());
+
+        List<OnmsAlarm> distinctOnmsAlarmList = new LinkedList<>();
+        Set<Integer> onmsAlarmIdSet = new TreeSet<>();
+
```

Comment:
The order has to be preserved. So this needs to be something 
like a linked list. Iterating over a map will destroy the sequence 
of the alarms.